core/src/forwarding_stage.rs

```diff
@@ -465,6 +507,86 @@ impl<VoteClient: ForwardingClient, NonVoteClient: ForwardingClient>
     }
 }
 
+#[async_trait]
+impl LeaderUpdater for ForwardAddressGetter {
+    fn next_leaders(&mut self, lookahead_slots: usize) -> Vec<SocketAddr> {
+        self.get_non_vote_forwarding_addresses(lookahead_slots as u64, Protocol::QUIC)
+    }
+
+    async fn stop(&mut self) {}
+}
+
+#[derive(Clone)]
+struct TpuClientNextClient {
+    runtime_handle: RuntimeHandle,
+    sender: mpsc::Sender<TransactionBatch>,
+}
+
+const METRICS_REPORTING_INTERVAL: Duration = Duration::from_secs(3);
+
+impl TpuClientNextClient {
+    fn new(
+        runtime_handle: tokio::runtime::Handle,
+        forward_address_getter: ForwardAddressGetter,
+        stake_identity: Option<&Keypair>,
+    ) -> Self {
+        // TODO(klykov): find the best channel size
+        let (sender, receiver) = mpsc::channel(16);
+        let cancel = CancellationToken::new();
+        let leader_updater = forward_address_getter.clone();
+
+        let config = Self::create_config(stake_identity, 1);
+        let scheduler = ConnectionWorkersScheduler::new(Box::new(leader_updater), receiver);
+        // leaking handle to this task, as it will run until the cancel signal is received
+        runtime_handle.spawn(scheduler.get_stats().report_to_influxdb(
+            "forwarding-stage-tpu-client",
+            METRICS_REPORTING_INTERVAL,
+            cancel.clone(),
+        ));
+        let _handle = runtime_handle.spawn(scheduler.run(config, cancel.clone()));
+        Self {
+            runtime_handle,
+            sender,
+        }
+    }
+
+    fn create_config(
+        stake_identity: Option<&Keypair>,
+        leader_forward_count: usize,
+    ) -> ConnectionWorkersSchedulerConfig {
+        ConnectionWorkersSchedulerConfig {
+            bind: SocketAddr::new(std::net::IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)), 0),
```

Comment:
do not bind to all interfaces, use bind_address from validator config.

Reply:
will check out if it is possible to carry this from the validator down here.

Reply:
if it is not possible we need to find a way. else this will blow up spectacularly on doublezero.

Reply:
It is possible but requires changes on the tpu-client-next first.

Reply:
Since these changes are done, I've updated this code as well.

Reply:
Thank you!